DotNetHowToEncryptionUsingCMK/DotNetHowToEncryptionUsingCMK/Program.cs

```diff
@@ -0,0 +1,329 @@
+                            LastRenovationDate = new DateTimeOffset(1982, 4, 28, 0, 0, 0, TimeSpan.Zero),
+                            Rating = 1,
+                            Location = GeographyPoint.Create(49.678581, -122.131577)
+                        }),
+                    IndexAction.MergeOrUpload(
+                        new Hotel()
+                        {
+                            HotelId = "3",
+                            BaseRate = 129.99,
+                            Description = "Close to town hall and the river"
+                        }),
+                    IndexAction.Delete(new Hotel() { HotelId = "6" })
+                };
+
+            var batch = IndexBatch.New(actions);
+
+            try
+            {
+                indexClient.Documents.Index(batch);
+            }
+            catch (IndexBatchException e)
+            {
+                // Sometimes when your Search service is under load, indexing will fail for some of the documents in
+                // the batch. Depending on your application, you can take compensating actions like delaying and
+                // retrying. For this simple demo, we just log the failed document keys and continue.
+                Console.WriteLine(
+                    "Failed to index some of the documents: {0}",
+                    String.Join(", ", e.IndexingResults.Where(r => !r.Succeeded).Select(r => r.Key)));
+            }
+
+            Console.WriteLine("Waiting for documents to be indexed...\n");
+            Thread.Sleep(2000);
+        }
+#endif
+
+        private static void RunQueries(ISearchIndexClient indexClient)
+        {
+            SearchParameters parameters;
+            DocumentSearchResult<Hotel> results;
+
+            Console.WriteLine("Search the entire index for the term 'budget' and return only the hotelName field:\n");
+
+            parameters =
+                new SearchParameters()
+                {
+                    Select = new[] { "hotelName" }
+                };
+
+            results = indexClient.Documents.Search<Hotel>("budget", parameters);
+
+            WriteDocuments(results);
+
+            Console.Write("Apply a filter to the index to find hotels cheaper than $150 per night, ");
+            Console.WriteLine("and return the hotelId and description:\n");
+
+            parameters =
+                new SearchParameters()
+                {
+                    Filter = "baseRate lt 150",
+                    Select = new[] { "hotelId", "description" }
+                };
+
+            results = indexClient.Documents.Search<Hotel>("*", parameters);
+
+            WriteDocuments(results);
+
+            Console.Write("Search the entire index, order by a specific field (lastRenovationDate) ");
+            Console.Write("in descending order, take the top two results, and show only hotelName and ");
+            Console.WriteLine("lastRenovationDate:\n");
+
+            parameters =
```

Comment:
To keep the narrative of this sample intact, the queries should focus on using synonyms to show that the encrypted synonym map still works.